### PR TITLE
Clk/donorlist

### PR DESCRIPTION
**Pull request by clukhei**
Two routes design in this PR

1)Donor list API 
sample API route: 
>/donors?from=2019-1-1&to=2020-4-26&taxDeduc=False&minAmt=6&maxAmt=10&source[]=Hackathon 2018&source[]=SUU President Challenge

This api is use to get donor records based on the filters. By default, we will let frontend decide the default date filters 

How this api works 
**Date filter**
1) Takes in date in the format YYYY-MM-DD
2) Mandatory to provide two dates (from and to ) 
- if no dates are provided, error message is thrown 

**Tax deductible filter  (taxDeduc)**
1) taxDeduc is optional 
2) boolean

**source filter (source[])**
1) source will be organised into an array by using source[] syntax in query params
2) there is no limit to the number of source the user can filter by 
3) source filter is optional

 **Donation amount filter (maxAmt , minAmt)**
1) total donation amount filter is applied last  before the response is generated (after all other filters are applied) 
2) donation amount filter is optional
3) user can filter by 
maxAmt only, 
minAmt only or 
maxAmt and minAmt 
4) it is not handled by sequelize 
5. Validation: minAmt must not be more than maxAmt, minAmt and maxAmt must be a number

--------------------------------------------------------------------------------------------------------------
2)Source API 
> /sources

This API is for FE to reflect sources of donation available in database,. 
Allows user to pick the sources that he/she wants to filter by

### CODE REVIEW

File: src/routes/donors.js [hunk 1]
```diff
@@ -1,155 +1,158 @@
...
-          [
-            Sequelize.fn('SUM', Sequelize.col('donationAmount')),
-            'totalAmountDonated',
-          ],
-        ],
+  if (from && to) {
+    donationConditions.donationDate = {
+      [Sequelize.Op.between]: [new Date(from), new Date(to)],
+    };
+  } else {
```
Comment (SheldonCheng) [inline src/routes/donors.js:32]: If the FE query doesn't include a date, then it should show all donors by default I think

Comment (clukhei) [reply src/routes/donors.js:32]: Hi @SheldonCheng , I was referring to thomas's comment in trello https://trello.com/c/SDG6dNjQ/43-donor-list-page-basic-filters

and was following up w kaleng, she told me the same that the default donor list page will always be first day of the year to today's date. Then its all up to the user to change the date filters if they want to see other date range

**Comment by shangong** [comment]
It looks good, although I haven't tested it. So only Donation amount filter (maxAmt , minAmt) is compulsory while Tax deductible filter (taxDeduc) and source filter (source[]) are optional?

**Comment by clukhei** [comment]
No.. only date filters are compulsory, and must provide to and from date.
The rest are optional .

**Comment by SheldonCheng** [comment]
@clukhei Would it be right to say that if FE does not provide date, then the date range should default to [start of current year] till [today]? If so then I think that default should just be put into BE rather than FE.

**Pull request merged**